Restore gdal build vrt alg, plus improvements

nyalldawson opened the pull request:
Brings back the GDAL build vrt algorithm, and adds usability improvements to the dialog for selecting layers to include. (e.g. you can now add layers which aren't part of the current project)

nyalldawson committed:
Port gdal build vrt to new API

nyalldawson committed:
Allow adding layers which aren't open in the project to processing
parameters which accept lists of multiple layers

E.g. build vrt alg, merge vector layers alg

Otherwise you may need to load 100's of layers temporarily into
a project to perform algs on them

nyalldawson committed:
Allow selecting multiple items in processing multi layer input dialog

And selecting/deselecting only these items. Otherwise it can be quite
cumbersome to manually select complex sets of items (e.g. try needing
to select ~50% of the layers in a large project - there's currently
no quick way to do this.) With this change you can at least
ctrl/shift click to create selections quicker.

nirvn commented:
@nyalldawson , I've tested this PR -- the output isn't automatically added when the output is is set to [ save as temporary file ]. Otherwise, all good. Thanks.

nyalldawson committed:
Default to vrt extension for build vrt alg output

nyalldawson committed:
Fix default layer sometimes inserted into multi layer parameter values

nyalldawson committed:
Handle mixed lists of values and lists when evaluating multilayer
parameters

This can happen when running models with mixed input types
for multilayer parameters

nyalldawson committed:
Fix multi layer parameters in models not giving all options for inputs

nyalldawson committed:
Fix exception when editing multi layer parameter in model

nyalldawson committed:
Allow adding manual layers in modeler multi layer alg parameters

This allows creation of models with child algorithms which do
things like merge a selected layer with a predefined static
layer, or create a vrt with a mix of static and user selected
layers.

nyalldawson commented:
>  I've tested this PR -- the output isn't automatically added when the output is is set to [ save as temporary file 

Should be mostly fixed now - the temporary output defaults to a vrt extension. A true solution relies on filtering the output raster layer available types to a subset of raster formats, but I'm not sure we want to expose that to the api yet (I'll see if any other use cases pop up first)
